crates/bevy_log/src/once.rs

```diff
@@ -0,0 +1,99 @@
+/// Call `trace!(...)` once per call site.
+///
+/// Useful for logging within systems which are called every frame.
+///
+/// Returns true the first time this is called
+#[macro_export]
+macro_rules! trace_once {
+    ($($arg:tt)+) => ({
+        use ::std::sync::atomic::{AtomicBool, Ordering};
+
+        static FIRST_TIME: AtomicBool = AtomicBool::new(true);
+        if FIRST_TIME.swap(false, Ordering::Relaxed) {
+            trace!($($arg)+);
```

Comment:
This expects `trace` to be in scope. You'll need to prefix with `$crate::`:
```suggestion
            $crate::trace!($($arg)+);
```

Same goes for the other macros.